src/lib/readable-stream/generic-reader.ts

```diff
@@ -2,10 +2,12 @@ import assert from '../../stub/assert';
 import { ReadableStream, ReadableStreamCancel, ReadableStreamReader } from '../readable-stream';
 import { newPromise, setPromiseIsHandledToTrue } from '../helpers';
 
-// TODO Fix ReadableStreamReadResult<R> in TypeScript DOM types
-export interface ReadResult<T = any> {
-  done: boolean;
+export type ReadResult<T> = {
```

Comment:
This aligns with TypeScript 3.8's [ReadableStreamReadDoneResult and ReadableStreamReadValueResult](https://github.com/microsoft/TypeScript/blob/v3.8.2/lib/lib.dom.d.ts#L1533-L1541).